# GitHub Repository Index

An index of my projects and tools. This document organizes repositories, gives short descriptions.

---

## Pinned repositories

- **Tool-Box** — *Utility scripts*  
  Useful scripts for Python and R, plus Docker, Snakemake, and Git.

- **Design-Pattern-Reference** — *Design pattern examples*  
  Illustrated examples of Creational, Structural, and Behavioral design patterns implemented across C++, Java, JavaScript, PHP, Python, and R.

- **NMFprofileR** — *R package: NMF + GO enrichment*  
  An R package that performs Non-negative Matrix Factorization (NMF) and gene ontology (GO) enrichment on NMF basis genes.

- **R_Shiny_App_TCGA_Subtype_KM_Plots** — *Interactive survival explorer*  
  A Shiny app that lets users select TCGA molecular subtypes and compare Kaplan–Meier survival curves.

- **openPIP** — *PPI browser / network visualizer*  
  Web framework for browsing and visualizing protein–protein interaction networks.

---

## Other repositories (alphabetical)
- **Bash_scripts** — Bash utility scripts
- **Canine_Lymphoma** — *Master’s project analysis* -  Analysis scripts and notebooks from the Master’s degree project focusing on canine lymphoma RNA-seq data.
- **GeneList-R** — R package for gene-list.com
- **GViz** — Code for visualizing genomic regions
- **ML** — Machine learning examples and notebooks
- **NMF** — Examples of NMF on BRCA TCGA dataset
- **Pharmaceuticals-Reference** — Information about marketed pharmaceuticals, indications, and molecular structures
- **REST-API-Reference** — Example of implementing a REST API in Python
- **Rosalind** — Solutions to problems on rosalind.info
- **scRNA-Seq** — Examples of scRNA-Seq analysis with Seurat and Scanpy
- **TCGA-Subtype-Reference** — Reference summarizing descriptions of TCGA molecular subtypes

---

## License & contact
MIT / Apache 2.0 recommended (miles [.] william [.] mee [@] gmail [.] com)

